src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlLanguageCharacteristics.cs

```diff
@@ -8,17 +8,33 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy
 {
     internal class HtmlLanguageCharacteristics : LanguageCharacteristics<HtmlTokenizer, HtmlSymbol, HtmlSymbolType>
     {
+        private bool _useDirectiveTokenizer;
         private static readonly HtmlLanguageCharacteristics _instance = new HtmlLanguageCharacteristics();
+        private static readonly HtmlLanguageCharacteristics _instanceWithDirectiveTokenizer =
```

Comment:
Are you able to just sublcass `HtmlLanguageCharacteristics` and then ovveride CreateTokenizer?